### ISSUE TITLE
[Hyper-V][Azure]cloudinit fail to get_source on Azure

### ISSUE THREAD

**Issue by ubuntu-server-builder** [canonical/cloud-init#2943]
This bug was originally filed in Launchpad as [LP: #1702256](https://bugs.launchpad.net/cloud-init/+bug/1702256)
<details>
<summary>Launchpad details</summary>
<pre>
affected_projects = []
assignee = redriver
assignee_name = Hongjiang Zhang
date_closed = None
date_created = 2017-07-04T09:26:48.547780+00:00
date_fix_committed = None
date_fix_released = None
id = 1702256
importance = undecided
is_complete = False
lp_url = https://bugs.launchpad.net/cloud-init/+bug/1702256
milestone = None
owner = redriver
owner_name = Hongjiang Zhang
private = False
status = in_progress
submitter = redriver
submitter_name = Hongjiang Zhang
tags = []
duplicates = []
</pre>
</details>

_Launchpad user **Hongjiang Zhang(redriver)** wrote on 2017-07-04T09:26:48.547780+00:00_

cloud-init on Azure cannot work properly:

This is in fact a regression, caused by commit: ebc9ecbc8a76bdf511a456fb72339a7eb4c20568

The issue happens in the following code: cloudinit/source/DataSourceAzure.py.

The removal of "sources.DEP_NETWORK" makes sources.list_sources() return [] for DataSourceAzure, which is the root cause.

+
 # Used to match classes to dependencies
 datasources = [
-    (DataSourceAzureNet, (sources.DEP_FILESYSTEM, sources.DEP_NETWORK)),
+    (DataSourceAzure, (sources.DEP_FILESYSTEM, )),
 ]

The error log:
2017-07-04 08:28:49,555 - __init__.py[DEBUG]: Searching for network data source in: []
2017-07-04 08:28:49,555 - util.py[WARNING]: No instance datasource found! Likely bad things to come!
2017-07-04 08:28:49,555 - util.py[DEBUG]: No instance datasource found! Likely bad things to come!
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/cloud_init-0.7.9-py2.7.egg/cloudinit/cmd/main.py", line 331, in main_init
    init.fetch(existing=existing)
  File "/usr/local/lib/python2.7/site-packages/cloud_init-0.7.9-py2.7.egg/cloudinit/stages.py", line 349, in fetch
    return self._get_data_source(existing=existing)
  File "/usr/local/lib/python2.7/site-packages/cloud_init-0.7.9-py2.7.egg/cloudinit/stages.py", line 259, in _get_data_source
    pkg_list, self.reporter)
  File "/usr/local/lib/python2.7/site-packages/cloud_init-0.7.9-py2.7.egg/cloudinit/sources/__init__.py", line 334, in find_source
    raise DataSourceNotFoundException(msg)
DataSourceNotFoundException: Did not find any data source, searched classes: ()

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Hongjiang Zhang(redriver)** wrote on 2017-07-04T09:57:41.346045+00:00_

This regression happens for cloudinit on FreeBSD.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Hongjiang Zhang(redriver)** wrote on 2017-07-04T10:39:41.987284+00:00_

It looks like "cloud-init init --local" does not work as expected. Let me check it further.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **David Britton(dpb)** wrote on 2017-07-05T14:35:08.491928+00:00_

Waiting on @redriver to get back.  Please re-open when ready.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Hongjiang Zhang(redriver)** wrote on 2017-07-06T02:14:47.563445+00:00_

I found the root cause: the script cloudinitlocal does not run because it has a circular dependence, which stifles the "cloud-init init --local". As a result, cloud-init fails for FreeBSD on Azure.

This issue was there, but the CI ebc9ecbc8a76bdf511a456fb72339a7eb4c20568 triggers the failure.

I have prepared the patch, and wait for view.

See https://code.launchpad.net/~redriver/cloud-init/+git/cloud-init/+merge/326905
